transaction.go

```diff
@@ -773,7 +797,12 @@ func (this *Transaction) SetTransactionValidDuration(duration time.Duration) *Tr
 }
 
 func (this *Transaction) GetTransactionID() TransactionID {
-	return this.transactionIDs._GetCurrent().(TransactionID)
+	if this.transactionIDs._Length() > 0 {
+		t := this.transactionIDs._GetCurrent().(TransactionID)
+		return t
+	}
+
+	return TransactionID{}
```

Comment:
is there a test case to prevent such regression?

Reply:
Just had it added to one of the new unit tests.